Makefile

```diff
@@ -0,0 +1,75 @@
+PROJECT_NAME := $(shell python setup.py --name)
+PROJECT_VERSION := $(shell python setup.py --version)
+
+BOLD := \033[1m
+RESET := \033[0m
+
+.PHONY: install
+install:
+	@echo "$(BOLD)Installing $(PROJECT_NAME) $(PROJECT_VERSION)$(RESET)"
+	@pip install .
+
+.PHONY: dev
+dev:
+	@echo "$(BOLD)Installing $(PROJECT_NAME) $(PROJECT_VERSION) in dev mode$(RESET)"
+	@pip install -e .[dev]
+
+.PHONY: check
+check: lint check-django check-migrations
+
+.PHONY: lint
+lint: check-black flake8 pylint mypy
+
+.PHONY: mypy
+mypy:
+	@echo "$(BOLD)Running mypy$(RESET)"
+	@mypy .
+
+.PHONY: check-black
+check-black:
+	@echo "$(BOLD)Checking black$(RESET)"
+	@black --check marsha/ 2>&1 | grep -Ev 'good job|All done'
+
+.PHONY: black
+black:
+	@echo "$(BOLD)Running black$(RESET)"
+	@black marsha/
+
+.PHONY: flake8
+flake8:
+	@echo "$(BOLD)Running flake8$(RESET)"
+	@mypy .
+
+.PHONY: pylint
+pylint:
+	@echo "$(BOLD)Running pylint$(RESET)"
+	@pylint marsha
+
+.PHONY: check-django
+check-django:
+	@echo "$(BOLD)Checking django$(RESET)"
+	@django-admin.py check
+
+.PHONY: check-migrations
+check-migrations:
+	@echo "$(BOLD)Checking migrations$(RESET)"
+	@django-admin.py makemigrations --check --dry-run
+
+.PHONY: doc
+doc:
+	@echo "$(BOLD)Building documentation$(RESET)"
+	@cd docs && $(MAKE) html
+
+.PHONY: dist
+dist:
+	@python setup.py sdist bdist_wheel
+
+.PHONY: clean
+clean:
+	@echo "$(BOLD)Cleaning$(RESET)"
+	@rm -rf build dist $(PROJECT_NAME).egg-info
+
+.PHONY: full-clean
+full-clean: clean
+	@echo "$(BOLD)Full cleaning$(RESET)"
+	@rm -rf .mypy_cache
```

Comment:
Can you add a help target (see [1]) and a description for each target to generate the help:

```Makefile
.PHONY: doc
doc: ## generate project's HTML documentation
	@echo "$(BOLD)Building documentation$(RESET)"
	@cd docs && $(MAKE) html
```

[1] https://github.com/openfun/richie/blob/master/Makefile#L105